libs/mimir/src/rubber.rs

```diff
@@ -632,27 +642,34 @@ impl Rubber {
     {
         use par_map::ParMap;
         use rs_es::operations::bulk::Action;
-        let mut nb = 0;
-        let chunk_size = 1000;
-        let chunks = iter.pack(chunk_size).par_map(|v| {
-            v.into_iter()
-                .map(|v| {
-                    v.es_id()
-                        .into_iter()
-                        .fold(Action::index(v), |action, id| action.with_id(id))
-                })
-                .collect::<Vec<_>>()
-        });
-        for chunk in chunks.filter(|c| !c.is_empty()) {
-            nb += chunk.len();
-            self.es_client
-                .bulk(&chunk)
-                .with_index(&index.name)
-                .with_doc_type(T::doc_type())
-                .send()?;
-        }
 
-        Ok(nb)
+        let chunk_size = 1000;
+        let index_name = index.name.to_owned();
+        let cnx_string = self.cnx_string.to_owned();
+        let timeout = self.timeout;
+
+        iter.pack(chunk_size)
+            .par_map(|v| {
+                v.into_iter()
+                    .map(|v| {
+                        v.es_id()
+                            .into_iter()
+                            .fold(Action::index(v), |action, id| action.with_id(id))
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .with_nb_threads(self.nb_insert_threads)
+            .par_map(move |chunk| {
+                rs_es::Client::init_with_timeout(&cnx_string, timeout)
+                    .expect("invalid ES connection string")
+                    .bulk(&chunk)
```

Comment:
Instead of duplicating the initialization, we could simply clone the existing  `es_client`.
(a bit ugly, admittedly)
```suggestion
            .par_map({
                let client = self.es_client.clone();
                move |chunk| {
                    client
                        .clone()
                        .bulk(&chunk)
                        ...
```

Reply:
That's actually better, for some reason I was convinced that the ES client wasn't `Sync`